### PR TITLE
iptables: 1.8.7 -> 1.8.8

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Update packages:
 - libnetfilter_cthelper to 1.0.1
 - libnetfilter_cttimeout to 1.0.1
 - iptables to 1.8.8

Changelog:
- http://nftables.org/projects/iptables/files/changes-iptables-1.8.8.txt

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
cc @SuperSandro2000

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Izorkin** [comment]
Added patch - https://git.netfilter.org/iptables/commit/?id=f319389525b066b7dc6d389c88f16a0df3b8f189
```
Gcc complains about missing types. Some commits introduced u_int* instead
of uint*. Use uint treewide.

Fixes errors in the form of:
In file included from xtables-legacy-multi.c:5:
xshared.h:83:56: error: unknown type name 'u_int16_t'; did you mean 'uint16_t'?
    83 | set_option(unsigned int *options, unsigned int option, u_int16_t *invflg,
        |                                                        ^~~~~~~~~
        |                                                        uint16_t
make[6]: *** [Makefile:712: xtables_legacy_multi-xtables-legacy-multi.o] Error 1

Avoid libipq API breakage by adjusting libipq.h include accordingly. For
arpt_mangle.h kernel uAPI header, apply same change as in kernel commit
e91ded8db5747 ("uapi: netfilter_arp: use __u8 instead of u_int8_t").
```

**Comment by Izorkin** [comment]
cc @SuperSandro2000 @misuzu @Mic92 @fpletz

File: pkgs/os-specific/linux/iptables/default.nix [hunk 1]
```diff
@@ -1,19 +1,35 @@
 { lib, stdenv, fetchurl, pkg-config, pruneLibtoolFiles, flex, bison
 , libmnl, libnetfilter_conntrack, libnfnetlink, libnftnl, libpcap
 , nftablesCompat ? true
+, fetchpatch
 }:
 
 with lib;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iptables/default.nix:7]: It would be nice to remove this.

Comment (Izorkin) [reply pkgs/os-specific/linux/iptables/default.nix:7]: fetchpatch is used

**Comment by Izorkin** [comment]
cc @ajs124

File: pkgs/os-specific/linux/iptables/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/iptables/default.nix:7]: I was talking about the `with lib` line.

Comment (Izorkin) [reply pkgs/os-specific/linux/iptables/default.nix]: Oh, sorry. Fixed.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by ajs124 — APPROVED** [review]
diff LGTM, but haven't tested this

**Comment by alyssais** [comment]
Those broke the iptables build on Musl.  Fix is #179570.

**Pull request merged**